src/coupling.jl

```diff
@@ -0,0 +1,134 @@
+SplitCoupledJumpProblem(prob::AbstractJumpProblem,prob_control::AbstractJumpProblem,aggregator::AbstractAggregatorAlgorithm,coupling_map::Vector{Tuple{Int64,Int64}})= JumpProblem(cat_problems(prob.prob,prob_control.prob),aggregator,build_split_jumps(prob,prob_control,coupling_map)...)
+
+# make new problem by joining initial_data
+function cat_problems(prob::DiscreteProblem,prob_control::DiscreteProblem)
+  u0_coupled = CoupledArray(prob.u0,prob_control.u0,true)
+  DiscreteProblem(u0_coupled,prob.tspan)
+end
+
+function cat_problems(prob::AbstractODEProblem,prob_control::AbstractODEProblem)
+  l = length(prob.u0) # add l_c = length(prob_control.u0)
+  new_f = function (t,u,du)
+    prob.f(t,u.u,@view du[1:l])
+    prob_control.f(t,u.u_control,@view du[l+1:2*l])
+  end
+  u0_coupled = CoupledArray(prob.u0,prob_control.u0,true)
+  ODEProblem(new_f,u0_coupled,prob.tspan)
+end
+
+function cat_problems(prob::DiscreteProblem,prob_control::AbstractODEProblem)
```

Comment:
Mixing these only makes sense if some non-defaults in the solver are on, otherwise a non-trivial `f` will act differently when changed to an `ODEProblem` (from a function map to an ODE RHS). This may cause problems. We may just want to disallow this.